camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/TagsYamlTest.java

```diff
@@ -0,0 +1,171 @@
+            }
+        }));
+        assertTrue(getTagsEventually(app).contains(BrooklynTags.newCampBrooklynTag("tag1")));
+    }
+
+    @Test
+    public void testBrooklynCampFailDslTags() throws Exception {
+        try {
+            final Entity app = createAndStartApplication(
+                    "services:",
+                    "- type: " + BasicApplication.class.getName(),
+                    "  brooklyn.tags:",
+                    "  - tag1",
+                    "  - $brooklyn:object:",
+                    "      type: "+TagsTestSerializableObject.class.getName(),
+                    "      constructor.args:",
+                    "      - $brooklyn:attributeWhenReady(\"host.name\")");
+            fail("Should throw IllegalArgumentException exception.");
+        } catch (CompoundRuntimeException e) {
+            Asserts.assertStringContainsAtLeastOne(Exceptions.getFirstInteresting(e).getMessage(),"brooklyn.tags should not contain DeferredSupplier objects, such example is using $brooklyn:attributeWhenReady");
+        }
+    }
+
+    @Test
+    public void testBrooklynCampFailNonSerializableDslTags() throws Exception {
+        try {
+            final Entity app = createAndStartApplication(
+                    "services:",
+                    "- type: " + BasicApplication.class.getName(),
+                    "  brooklyn.tags:",
+                    "  - tag1",
+                    "  - $brooklyn:object:",
+                    "      type: "+NonSerializable.class.getName());
+            fail("Should throw IllegalArgumentException exception.");
+        } catch (CompoundRuntimeException e) {
+            Asserts.assertStringContainsAtLeastOne(Exceptions.getFirstInteresting(e).getMessage(),"brooklyn.tags should not contain DeferredSupplier objects, such example is using $brooklyn:attributeWhenReady");
+        }
+    }
+
+    public static class NonSerializable {}
+
+    @Test
+    public void testTagWithDslValue() throws Exception {
+        Entity app = createAndStartApplication(
+                "services:",
+                "- type: " + BasicApplication.class.getName(),
+                "  brooklyn.tags:",
+                "  - $brooklyn:literal(\"myval\")");
+        assertTrue(getTagsEventually(app).contains(BrooklynTags.newCampBrooklynTag("myval")));
+    }
+
+    public static class TagsTestSerializableObject implements Serializable {
+        private static final long serialVersionUID = -5150802977330637460L;
+
+        public TagsTestSerializableObject() {}
+        public TagsTestSerializableObject(Object arg1) {}
+    }
+
+    private static Set<Object> getTagsEventually(final Entity entity) throws Exception {
+        Task<Set<Object>> result = ((EntityInternal)entity).getExecutionContext().submit(new Callable<Set<Object>>() {
+            @Override
+            public Set<Object> call() throws Exception {
+                return entity.tags().getTags();
```

Comment:
Why do we need to call this inside a task? Does it fail without that? Under what circumstances would `entity.tags().getTags()` block, and timeout?

Why do we need "eventually" - is the tag potentially not there initially?